### ISSUE TITLE
Variants from same position but on different strant

### ISSUE THREAD

**Issue by kokyriakidis** [KarchinLab/open-cravat#76]
Hi!

How do you manage variants on different strands? For example A->G (+ strand) and T->C (- strand) in the same position. I want to merge some vcf files that have these kind of behavior and I wonder if I need to convert everything to the + strand before using opencravat.

**Comment by rkimoakbioinformatics** [comment]
Hi @kokyriakidis, OpenCRAVAT expects VCF-format input files to follow the VCF format standard which is that all variants are written from the + strand's viewpoint.

With that said, do you mean that some of your VCF-format input files have "chr1 123456 A G" and others have "chr1 123456 T C"? Do they come from some pipeline?

**Comment by kokyriakidis** [comment]
> With that said, do you mean that some of your VCF-format input files have "chr1 123456 A G" and others have "chr1 123456 T C"? Do they come from some pipeline?

Yes I mean exactly that. I use a pipeline to detect RNA-Editing events. It outputs a tab seperated file with rna candidate positions. Some may be from - strand some from + strand. I tried to convert them manually to a vcf file in order to be able to annotate them using open-cravat. But when I tried to merge the vcfs from all the samples I got a warning from bcftools that "The REF prefixes differ". This is expected because some samples have variation in the same position but in different strand.

In strand specific protocols RNA Events in humans should be A->G in the + strand and T->C in the - strand. So I guess I should convert everything from the - strand to the + strand.

**Comment by rkimoakbioinformatics** [comment]
Thanks for the information. Yes, at this moment, all variants in a VCF-format file should be based on the + strand. 

I can think of adding a "correction" feature to OpenCRAVAT, so that it can check the reference base and convert if necessary, but I'd like to check how indels are written by your pipeline, especially the orientation of sequences. For example, 

1. Insertion of AG between chr1:10000T and chr1:10001C (TC > TAGC on the + strand).
1.1 + strand notation: chr1 10000 T TAG
1.2 - strand notation: chr1 10001 G GCT

2. Deletion of AG between chr1:1000T and chr1:10003C (TAGC > TC on the + strand).
2.1 + strand notation: chr1 10000 TAG T
2.2 - strand notation: chr1 10003 GTC G

Are 1.2 and 2.2 what happen by your pipeline? If 1.2 is written as chr1 10000 A ATC for example it would be misleading.

**Comment by kokyriakidis** [comment]
Hmm, RNA Editing should not cause any INS OR DEL so I guess I am not the right person to answer something like that. RNA-Editing pipelines output only SNVs.

**Comment by rkimoakbioinformatics** [comment]
@kokyriakidis Yes. Thanks for letting us know about the database. We'll take a look and keep you updated.

Regarding the VCF format issue, in that case, automatic conversion may be possible. We'll keep you updated about it, too.

**Comment by Juke34** [comment]
@kokyriakidis awk+sed command should be enough to convert your - strand VCF file into + strand VCF file. Do you work with a mixed VCF file? or one that represent the + strand and another the - strand?
In the first case you should revise your workflow to handle both cases in parallel, convert the - strand file and then merge into a single + strand file.

**Comment by kokyriakidis** [comment]
@Juke34 I work with a mixed VCF file and did wat you proposed. Converted the - strand inputs to + strand. AtoG in the - strand was converted to TtoC in the + strand.
